### PR TITLE
--json= argument rework in various tools

### PR DESCRIPTION
**Pull request by poettering**
Let's simplify/unify the code that parses the --json= command line parameter. THis has bugged me for a while, let's do something about it before we end up copying the same code into even more tools.

### CODE REVIEW

File: src/dissect/dissect.c [hunk 1]
```diff
@@ -353,17 +339,17 @@ static int action_dissect(DissectedImage *m, LoopDevice *d) {
         assert(m);
         assert(d);
 
-        if (!arg_json)
+        if (arg_json_format_flags & JSON_FORMAT_OFF)
```
Comment (bluca) [inline src/dissect/dissect.c:342]: nit: when are we supposed to use FLAGS_SET and when not? It seems a bit incosistent through the code base

File: src/busctl/busctl.c [hunk 2]
```diff
@@ -2543,25 +2527,13 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
 
                 case 'j':
-                        if (on_tty())
-                                arg_json = JSON_PRETTY;
-                        else
-                                arg_json = JSON_SHORT;
+                        arg_json_format_flags = JSON_FORMAT_PRETTY_AUTO|JSON_FORMAT_COLOR_AUTO;
```
Comment (bluca) [inline src/busctl/busctl.c:2530]: Is JSON_SHORT the equivalent of JSON_FORMAT_NEWLINE? Will the output be the same?


```
if (((flags & (JSON_FORMAT_PRETTY_AUTO|JSON_FORMAT_PRETTY)) == JSON_FORMAT_PRETTY_AUTO))
    flags |= on_tty() ? JSON_FORMAT_PRETTY : JSON_FORMAT_NEWLINE;
```

**Review by bluca — APPROVED** [review]
Just 2 minor questions, but looks like a good cleanup - lgtm

File: src/dissect/dissect.c (hunk 1 above)
Comment (poettering) [reply src/dissect/dissect.c:342]: FLAGS_SET() matters in particular when you test flags that are combinations of bits, or multiple flags at once. For the rest it's up to you I guess. I tend to use the syntax `if (x & flag)` for simple flag check, but when checking if a bit is off i use `if (!FLAGS_SET(x, flag))` right now, since i found `if (!(x & flag))` so ugly, because of the `(!(` part, but I don't expect anyone to agree with me on that... ;-)

basically, it's up to you, but if you spell out `x & (f1|f2) == (f1|f2)` then you better use `FLAGS_SET(x, f1|f2)`.

the macro is called "FLAG*S*" after all, i.e. is in plural, pointing to the fact that it is useful when checking multiple flags at once

File: src/busctl/busctl.c (hunk 2 above)
Comment (poettering) [reply src/busctl/busctl.c:2530]: The old `JSON_SHORT` was equivalent to `JSON_FORMAT_NEWLINE|JSON_FORMAT_COLOR_AUTO`, hence yes, the output should not change by this. (see the json_dump_with_flags() function that is removed further up fo the translation of JSON_SHORT to the flags.

**Comment by poettering** [comment]
thanks for the review, let's merge.
